### ISSUE TITLE
Unable to test functions with camel-case names

### ISSUE THREAD

**Issue by fatmcgav** [rodjek/rspec-puppet#388]
I appear to have identified an issue with rspec-puppet failing when attempting to test functions with camel-case names.

To illustrate the issue, I've stuck a very simple sample repo up here: https://github.com/fatmcgav/puppet-function-test
And stuck travis-ci on it aswell. An example of the failure is here: https://travis-ci.org/fatmcgav/puppet-function-test/jobs/130016668

The error I'm seeing is:

> ```
>  NoMethodError:
>    undefined method `name' for nil:NilClass
> ```

Let me know if I can provide any more info... 

Cheers
Gavin

**Comment by clabu** [comment]
It is also not possible to test camel-case parameters:

`
it { should contain_myresource('a').with_camelCase(true) }
`

**Comment by rodjek** [comment]
@fatmcgav this has been fixed in #470.

@clabu While camel case attribute names might work for certain versions of Puppet, the use of capital letters in attribute names is not a supported part of the language [according to the documentation](https://docs.puppet.com/puppet/4.9/lang_resources.html#simplified-syntax), so it's unlikely that we'll be adding support for it.

```
Optionally, any number of attribute and value pairs, each of which consists of:
    * An attribute name, which is a lowercase word with no quotes.
```
